File: javascript/numeric_spec.js
var should = require('chai').should();
var Algos = require('./numeric');

var Ary = function () {
  return [1, 2, 3];
};

describe('numeric algorithms', function(){
  
  it('itoa', function(){
    var g =  Algos.itoa(3);
    g().should.be.equal(3);
    g().should.be.equal(4);
  });

  it('accumulate', function(){
    Algos.accumulate(Ary(), 1, function(x, y) {
      return x * y;
    }).should.be.equal(6);
  });

  it('inner_production', function(){
    Algos.inner_production(Ary(), Ary(), 0).should.be.equal(14);
  });

  it('adjacent_difference', function(){
    JSON.stringify(Algos.adjacent_difference(Ary()))
      .should.be.equal(JSON.stringify([1, 1, 1]));
  });

  it('partial_sum', function(){
    JSON.stringify(Algos.partial_sum(Ary()))
      .should.be.equal(JSON.stringify([1, 3, 6]));
  });
});

File: lua/numeric_spec.lua
local Algos = require('./numeric')

local ary = {1, 2, 3}

describe('numeric algorithms', function()
    it('itoa', function()
        local g = Algos.itoa(3)
        assert.are.same(3, g())
        assert.are.same(4, g())
    end)

    it('accumulate', function()
        assert.are.same(1, Algos.accumulate({}, 1, function(x, y)
            return x * y
        end))
        assert.are.same(6, Algos.accumulate(ary, 1, function(x, y)
            return x * y
        end))
    end)

    it('innerProduction', function()
        assert.are.same(1, Algos.innerProduction({}, {}, 1))
        assert.are.same(14, Algos.innerProduction(ary, ary, 0))
    end)

    it('adjacentDifference', function()
        assert.are.same({}, Algos.adjacentDifference({}))
        assert.are.same({1, 1, 1}, Algos.adjacentDifference(ary))
    end)

    it('partialSum', function()
        assert.are.same({}, Algos.partialSum({}))
        assert.are.same({1, 3, 6}, Algos.partialSum(ary))
    end)
end)
